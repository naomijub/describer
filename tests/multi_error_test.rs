use describer::Describe;

#[derive(Describe)]
#[prettify(token(separator = ",", spacing = "\n", keyval = "=>"))]
#[prettify(tokens(sepurator = ",", spacigg = "\n", key_val = "=>"))]
#[prettify(hide_opt = 1)]
#[prettify(explicit_collections = "true")]
#[prettify(hide_optional = true)]
struct MyStruct {
    vec_u8s: Vec<u8>,
    other: u8,
}
 
fn main() {
    assert_eq!(
        MyStruct::describe(),
        "MyStruct\n{\nvec_u8s=>[u8!]!,other=>u8!\n}"
    );
}