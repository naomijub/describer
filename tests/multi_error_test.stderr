error: expected `,`
 --> tests/multi_error_test.rs:5:29
  |
5 | #[prettify(tokens(sepurator = ",", spacigg = "\n", key_val = "=>"))]
  |                             ^

error: `hide_opt` supports only boolean types
 --> tests/multi_error_test.rs:6:23
  |
6 | #[prettify(hide_opt = 1)]
  |                       ^

error: `explicit_collections` supports only boolean types
 --> tests/multi_error_test.rs:7:35
  |
7 | #[prettify(explicit_collections = "true")]
  |                                   ^^^^^^

error[E0599]: no function or associated item named `describe` found for struct `MyStruct` in the current scope
  --> tests/multi_error_test.rs:16:19
   |
9  | struct MyStruct {
   | --------------- function or associated item `describe` not found for this struct
...
16 |         MyStruct::describe(),
   |                   ^^^^^^^^ function or associated item not found in `MyStruct`
